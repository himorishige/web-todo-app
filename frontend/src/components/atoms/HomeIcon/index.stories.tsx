/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import HomeIcon from '.';

export default {
  title: 'WebToDoApp/Atoms/HomeIcon',
  component: HomeIcon,
} as ComponentMeta<typeof HomeIcon>;

const Template: ComponentStory<typeof HomeIcon> = (args) => <HomeIcon {...args} />;

export const Default = Template.bind({});
Default.storyName = 'PriorityButton';
