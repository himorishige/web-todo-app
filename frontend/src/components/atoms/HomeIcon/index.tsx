/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { memo } from 'react';
import { BsTable } from 'react-icons/bs';

const HomeIcon: React.VFC = memo(() => {
  return (
    <div>
      <BsTable css={iconStyle} />
    </div>
  );
});

export default HomeIcon;

const iconStyle = css`
  width: 2rem;
  height: 2rem;
  margin-right: 0.5rem;
`;
