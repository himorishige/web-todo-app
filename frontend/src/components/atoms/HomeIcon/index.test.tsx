import renderer from 'react-test-renderer';
import HomeIcon from '.';

describe('Atoms/HomeIcon', () => {
  test('正しくレンダリングされている', () => {
    const target = renderer.create(<HomeIcon />).toJSON();
    expect(target).toMatchSnapshot();
  });
});
