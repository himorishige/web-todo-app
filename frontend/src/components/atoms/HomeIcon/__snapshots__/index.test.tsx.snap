// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/HomeIcon 正しくレンダリングされている 1`] = `
<div>
  <svg
    className="css-1443oa1"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 16 16"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clipRule="evenodd"
      d="M14 1H2a1 1 0 00-1 1v12a1 1 0 001 1h12a1 1 0 001-1V2a1 1 0 00-1-1zM2 0a2 2 0 00-2 2v12a2 2 0 002 2h12a2 2 0 002-2V2a2 2 0 00-2-2H2z"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M15 4H1V3h14v1z"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M5 15.5v-14h1v14H5zm5 0v-14h1v14h-1z"
      fillRule="evenodd"
    />
    <path
      clipRule="evenodd"
      d="M15 8H1V7h14v1zm0 4H1v-1h14v1z"
      fillRule="evenodd"
    />
    <path
      d="M0 2a2 2 0 012-2h12a2 2 0 012 2v2H0V2z"
    />
  </svg>
</div>
`;
